### PR TITLE
feat(medium): init component

### PR DESCRIPTION
**Pull request by manoncarbonnel**
Adding a new component called Medium

### CODE REVIEW

File: packages/components/medium/src/components/osds-medium/core/controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+class OdsMediumMock extends OsdsMedium {
+  constructor(attribute: Partial<OsdsMedium>) {
+    super();
+    Object.assign(this, attribute);
+  }
+
+  controller: OdsMediumController = jest.fn() as unknown as OdsMediumController;
+  odsRemoveClick!: EventEmitter<void>;
+  removeClicked = jest.fn();
```
Comment (dpellier) [inline packages/components/medium/src/components/osds-medium/core/controller.spec.ts:17]: remove

**Review by dpellier — CHANGES_REQUESTED** [review]
no need to put number on file names in `packages/storybook/stories/components/medium`

**Review by dpellier — CHANGES_REQUESTED** [review]
Please update storybook controls so that we can update width and height
Capture d’écran 2023-11-08 à 17 13 48

File: packages/components/medium/src/components/osds-medium/core/controller.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  constructor(component: OsdsMedium) {
+    this.component = component;
+  }
+
+  /**
+   * Warn user if the alt attribute is not set
+   * @param alt - alternative description of the medium
+   */
+  validateAlt(alt: string): void {
+    if (alt.length == 0) {
```
Comment (aesteves60) [inline packages/components/medium/src/components/osds-medium/core/controller.ts:21]:
```suggestion
    if (alt.length === 0) {
```

File: packages/components/medium/src/components/osds-medium/core/controller.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      this.logger.warn('The alt attribute must be set');
+    }
+  }
+
+  /**
+   * Warn user if the src attribute is not set
+   * @param src - url of the medium
+   */
+  validateSrc(src: string): void {
+    if (src.length == 0) {
```
Comment (aesteves60) [inline packages/components/medium/src/components/osds-medium/core/controller.ts:31]:
```suggestion
    if (src.length === 0) {
```

File: packages/components/medium/src/components/osds-medium/osds-medium.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'osds-medium.scss',
+  tag: 'osds-medium',
+})
+export class OsdsMedium implements OdsMediumAttribute {
+  controller: OdsMediumController = new OdsMediumController(this);
+
+  /** @see OdsMediumAttributes.alt */
```
Comment (aesteves60) [inline packages/components/medium/src/components/osds-medium/osds-medium.tsx:18]: I think this type of comment are useless, not utils in the doc generated

File: packages/components/medium/src/components/osds-medium/osds-medium.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  connectedCallback(): void {
+    this.beforeInit();
+  }
+
+  render(): FunctionalComponent {
+    return (
+      <Host>
+        <img alt={ this.alt }
+          src={ this.src }
+          {...(this.height ? { height: this.height } : {}) }
```
Comment (aesteves60) [inline packages/components/medium/src/components/osds-medium/osds-medium.tsx:57]: Don't work with a undefined fallback ? For avoid the double deconstruction
like that `height={this.height || undefined}`

**Review by aesteves60 — CHANGES_REQUESTED** [review]

File: packages/components/medium/src/components/osds-medium/osds-medium.tsx (hunk 5 above)
Comment (dpellier) [reply packages/components/medium/src/components/osds-medium/osds-medium.tsx]: Technically `0` should be a valid value, in that case it won't be

Comment (dpellier) [reply packages/components/medium/src/components/osds-medium/osds-medium.tsx]: use `??` instead

**Review by dpellier — APPROVED** [review]

**Review by Leotheluck — APPROVED** [review]

**Review by aesteves60 — APPROVED** [review]

**Pull request merged**